source/modes/filebrowser.c

```diff
@@ -150,7 +150,14 @@ static gint compare_name(gconstpointer a, gconstpointer b,
     return fa->type - fb->type;
   }
 
-  return g_strcmp0(fa->name, fb->name);
+  const gchar *fa_key = g_utf8_collate_key_for_filename(fa->name, -1);
+  const gchar *fb_key = g_utf8_collate_key_for_filename(fb->name, -1);
+
+  gint order = g_strcmp0(fa_key, fb_key);
+
+  g_free(fa_key);
```

Comment:
It allocates and frees for every of the n*log(n) comparisons which looks wasteful.

Not sure if it has a noticeable effect on large directories but it's worth checking.